File: Frontend/src/types/project.ts
export type ProjectStatus = 'live' | 'in-progress' | 'completed' | 'research' | 'por-definir'

export interface Project {
  id: string
  title: string
  description: string
  problem: string
  solution: string
  technologies: string[]
  role: string
  status: ProjectStatus
  image?: string
  demoUrl?: string
  githubUrl?: string
  featured: boolean
}

File: Backend/app/schemas/project.py
from pydantic import BaseModel


class Project(BaseModel):
    id: str
    title: str
    description: str
    problem: str
    solution: str
    technologies: list[str]
    role: str
    status: str
    image: str | None = None
    demo_url: str | None = None
    github_url: str | None = None
    featured: bool = False
